src/shogun/optimization/DescendCorrection.h

```diff
@@ -68,11 +68,12 @@ class DescendCorrection
 	 *
 	 * @param negative_descend_direction the negative descend direction
 	 * @param idx the index of the direction
+	 * @param delta update the change to correct descend direction
 	 * 
 	 * @return corrected descend direction
 	 */
 	virtual float64_t get_corrected_descend_direction(float64_t negative_descend_direction,
-		index_t idx)=0;
+		index_t idx, float64_t& delta)=0;
```

Comment:
The idea of Adaptive momentum is 
* increase the gradient if the sign of `velocity` and the sign of `gradient` are agreed
* discount the gradient  if the sign of `velocity` and the sign of `gradient` are not agreed